### PR TITLE
Clear VCS fsbackend and git_pillar locks on master start

### PR DESCRIPTION
**Pull request by terminalmage**
This is an alternate solution for the issue raised in #32314.

### CODE REVIEW

**Comment by terminalmage** [comment]
This could use clarification in the documentation that if a lock is present, it will be cleared when the master restarts.

**Comment by terminalmage** [comment]
@thatch45 Could you look at the ioflo stuff that I added for this? I've never touched ioflo code and had to make those changes based on intuition.

File: salt/daemons/flo/maint.flo [hunk 1]
```diff
@@ -16,6 +16,14 @@ framer jobcleaner be active first setup
         enter
             do salt raet maint fileserver clean
         go start
+    frame clearfslocks
+        enter
+            do salt raet maint fileserver clear locks
+        go start
```
Comment (thatch45) [inline salt/daemons/flo/maint.flo]: This should be:
go cleargitpillarlocks

File: salt/daemons/flo/maint.flo [hunk 2]
```diff
@@ -16,6 +16,14 @@ framer jobcleaner be active first setup
         enter
             do salt raet maint fileserver clean
         go start
```
Comment (thatch45) [inline salt/daemons/flo/maint.flo]: should be:
go clearfslocks


Otherwise these will never be called

File: salt/daemons/flo/maint.flo (hunk 1 above)
Comment (terminalmage) [inline salt/daemons/flo/maint.flo]: Thanks, I updated both.

**Comment by thatch45** [comment]
Perfect!

**Pull request merged**